backend/src/main/java/fk/prof/backend/model/policy/ZookeeperBasedPolicyStoreAPI.java

```diff
@@ -0,0 +1,193 @@
+        if (processGroup == null) {
+            throw new IllegalArgumentException("Process group is required");
+        }
+        if (requestedVersionedPolicyDetails == null) {
+            throw new IllegalArgumentException("PolicyDetails is required");
+        }
+        Future<PolicyDTO.VersionedPolicyDetails> future = Future.future();
+        String policyNodePath = PathNamingUtil.getPolicyNodePath(processGroup, policyPath, policyVersion);
+
+        vertx.executeBlocking(fut -> {
+            try {
+                PolicyDTO.VersionedPolicyDetails newVersionedPolicy = policyCache.compute(processGroup, (k, v) -> {
+                    if (v != null && create) {
+                        throw new PolicyException(String.format("Failing create of policy, Policy for ProcessGroup = %s already exists, policyDetails = %s", RecorderProtoUtil.processGroupCompactRepr(processGroup), PolicyProtoUtil.versionedPolicyDetailsCompactRepr(getVersionedPolicy(processGroup))), true);
+                    }
+                    if (v == null && !create) {
+                        throw new PolicyException(String.format("Failing update of policy, Policy for ProcessGroup = %s does not exist", RecorderProtoUtil.processGroupCompactRepr(processGroup)), true);
+                    }
+                    if (v != null && v.getVersion() != requestedVersionedPolicyDetails.getVersion()) {
+                        throw new PolicyException("Failing update of policy, policy version mismatch, current version = " + v.getVersion() + ", your version = " + requestedVersionedPolicyDetails.getVersion() + ", for ProcessGroup = " + RecorderProtoUtil.processGroupCompactRepr(processGroup), true);
+                    }
+                    try {
+                        String res = curatorClient.create().creatingParentsIfNeeded().withMode(CreateMode.PERSISTENT_SEQUENTIAL).
+                                forPath(policyNodePath, requestedVersionedPolicyDetails.getPolicyDetails().toByteArray());
+                        res = ZKPaths.getNodeFromPath(res);
+                        Integer newVersion = Integer.parseInt(res);
+
+                        PolicyDTO.VersionedPolicyDetails updated = requestedVersionedPolicyDetails.toBuilder().setVersion(newVersion).build();
+                        putInProcessGroupAsTree(processGroup);
+                        return updated;
+                    } catch (Exception e) {
+                        throw new PolicyException("Exception thrown by ZK while writing policy for ProcessGroup = " + RecorderProtoUtil.processGroupCompactRepr(processGroup), e, true);
+                    }
+                });
+                fut.complete(newVersionedPolicy);
+            } catch (Exception e) {
+                fut.fail(e);
+            }
+        }, true, future.completer());
```

Comment:
remove ordered constraint if not required. this impacts perf of all execute blocking calls on this event loop (of other APIs as well)